### PR TITLE
Fetch Repositories

### PR DESCRIPTION
**Pull request by KBandipo**
This pull request makes the following changes 

- Fixes (https://discord.com/channels/1148391778492354741/1253584233155395657/1253594458172297270)

Add functionality to fetch the list of all public repositories on GitHub, alongside their repository links using Get request and display them on the UI upon page load.

### CODE REVIEW

**Comment by KBandipo** [comment]
Hello @collabo-community/review-web-app , my PR is ready for review. Thank you

File: .env.example [hunk 1]
```diff
@@ -5,4 +5,7 @@ PORT=
 GITHUB_PERSONAL_ACCESS_TOKEN=
 
 # Supply base URL
-GITHUB_API_URL=
\ No newline at end of file
+GITHUB_API_URL=
+
+# GITHUB REPOSITORIES API URL
+GITHUB_REPOS_API_URL=https://api.github.com/user/repos
```
Comment (Ifycode) [inline .env.example:11]: Hmm... Let's see if we can reuse variables inside of the .env file @KBandipo
- See stackoverflow comment: https://stackoverflow.com/a/67615097
- Additional resource just incase: https://stackoverflow.com/a/48794954

**Review by Ifycode — CHANGES_REQUESTED** [review]

**Comment by KBandipo** [comment]
My pull request is ready for review. I have reused the base url variable inside of the .env file. This is the link to the task issue https://discord.com/channels/1148391778492354741/1253584233155395657/1253594458172297270
Thank you

File: .env.example [hunk 2]
```diff
@@ -4,5 +4,12 @@ PORT=
...
-GITHUB_API_URL=
\ No newline at end of file
+# Github Base URL
+BASE_URL=https://api.github.com
+
+# Supply your own github account user name and the name of the repo you want to interact with
+GITHUB_API_URL=${BASE_URL}/repos/your-own-github-account-user-name-will-be-here-instead/the-name-of-the-repo-you-want-to-interact-with
+
+# Github Repositories API URL 
+GITHUB_REPOS_API_URL=${BASE_URL}/user/repos
```
Comment (Ifycode) [inline .env.example:14]: You can just remove this one from here entirely, then you can use it in the code directly like this:

`${process.env.GITHUB_API_BASE_URL}/user/repos`

**Review by Ifycode — CHANGES_REQUESTED** [review]
Nice Work @KBandipo 
Let's just clean it up a bit. See comments below.

**Review by Ifycode — CHANGES_REQUESTED** [review]
@KBandipo always link your issue to your PR. Add your issue link approriately.

**Comment by Ifycode** [comment]
@KBandipo I have the same question to ask you. the one that I asked Daniel. 
The link to your issue is in the wrong place. How do you expect reviewers/users to find it where you currently added it?

File: package-lock.json [hunk 3]
```diff
@@ -10,10 +10,10 @@
       "license": "ISC",
       "dependencies": {
```
Comment (Ifycode) [inline package-lock.json:11]: package-lock.json should not be showing in your file changes @KBandipo

**Review by Ifycode — CHANGES_REQUESTED** [review]

File: .env.example [hunk 4]
```diff
@@ -4,5 +4,12 @@ PORT=
 # Provide your GitHub Personal access token
 GITHUB_PERSONAL_ACCESS_TOKEN=
 
-# Supply base URL
-GITHUB_API_URL=
\ No newline at end of file
+# Github Base URL
+BASE_URL=https://api.github.com
```
[the remark this answers is not in the document]
Comment (KBandipo) [reply .env.example:8]: Okayyyy

File: .env.example [hunk 5]
```diff
@@ -4,5 +4,12 @@ PORT=
...
+
+# Supply your own github account user name and the name of the repo you want to interact with
+GITHUB_API_URL=${BASE_URL}/repos/your-own-github-account-user-name-will-be-here-instead/the-name-of-the-repo-you-want-to-interact-with
```
[the remark this answers is not in the document]
Comment (KBandipo) [reply .env.example:11]: Copy boss

File: package-lock.json (hunk 3 above)
Comment (KBandipo) [reply package-lock.json:11]: Alright, @Ifycode  please how can I restore it?

**Comment by KBandipo** [comment]
Howdy boss, I have restored the package-lock.json as you requested @collabo-community/review-collabocate . I'm still working on the other things you corrected too. Thank you

**Comment by KBandipo** [comment]
Hello @collabo-community/review-collabocate , my pull request is ready for review. These are my changes as you have requested; I replaced `BASE_URL` with `GITHUB_API_BASE_URL` and `GITHUB_API_URL` to `REPO_URL` in `.env.example`;  and Refactored the usage of Github Repositories API URL to `process.env.GITHUB_API_BASE_URL` directly in the server.js

**Comment by tylerjusfly** [comment]
It says that you have a conflict, can you resolve that @KBandipo ?

**Comment by KBandipo** [comment]
@tylerjusfly Thank you boss, I may not be able to resolve the conflict myself since I am not the only one working on the old server. @Ifycode is managing the issues together

**Review by Ifycode — CHANGES_REQUESTED** [review]
@KBandipo you are yet to link your issue to the PR properly, I'm still waiting for this before I can touch/review your PR.

#

> About conflicts, @tylerjusfly I'll be the one to resolve the conflicts (it's caused by the restructuring I'm doing based all our 4 different PRs), so @KBandipo should not worry about the conflicts in this case

**Comment by KBandipo** [comment]
Thank you @Ifycode , I have linked the issue to the pull request now. Please let me know if there are any additional steps I need to take

**Review by Ifycode — APPROVED** [review]

**Comment by Ifycode** [comment]
@all-contributors add @KBandipo for code

**Pull request merged**